### PR TITLE
fix(medusa-payment-stripe): Catch on idempotency key retrieve if not found

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
In case the idempotency key is not found, an error will be thrown. We actually want to catch this error and create the key if it does not exists

### CODE REVIEW

File: packages/medusa-payment-stripe/src/api/utils/__tests__/utils.spec.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ describe("Utils", () => {
   describe("handlePaymentHook", () => {
     describe("on event type payment_intent.succeeded", () => {
       describe("in a payment context", () => {
-        it("should complete the cart on non existing order", async () => {
+        it.only("should complete the cart on non existing order", async () => {
```
Comment (olivermrbl) [inline packages/medusa-payment-stripe/src/api/utils/__tests__/utils.spec.ts]:
```suggestion
        it("should complete the cart on non existing order", async () => {
```

File: packages/medusa-payment-stripe/src/api/utils/__tests__/utils.spec.ts [hunk 2]
```diff
@@ -244,7 +244,7 @@ describe("Utils", () => {
     })
 
     describe("on event type payment_intent.amount_capturable_updated", () => {
-      it("should complete the cart on non existing order", async () => {
+      it.only("should complete the cart on non existing order", async () => {
```
Comment (olivermrbl) [inline packages/medusa-payment-stripe/src/api/utils/__tests__/utils.spec.ts]:
```suggestion
      it("should complete the cart on non existing order", async () => {
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Review by olivermrbl — APPROVED** [review]
LGTM

**Comment by pepijn-vanvlaanderen** [comment]
@adrien2p we came accross this issue as well, your PR fixes the retrieval when no idempotency key exists, however there is also a missing withTransaction in the `capturePaymentIfNecessary` function, resulting in no order found when it is just created before. Also see this PR of my co-worker: https://github.com/medusajs/medusa/pull/3930/files#diff-a690d99fec6c0969d14e9ff67236a1ee53e1b694f4bac7e3b22bb7a51aca638c

**Pull request merged**